### ISSUE TITLE
Support `using Statistics as Stat`

### ISSUE THREAD

**Issue by LilithHafner** [JuliaLang/julia#52784]
I propose that `using Statistics as Stat` loads all exported names in Statistics but does not load the name Statistics itself, instead loading the module as the name `Stat`.

This almost is equivalent to 
```julia
using Statistics
using Statistics: Statistics as Stat
```
And is equivalent to
```julia
using Statistics: Statistics as Stat
eval(Expr(:using, Expr(:(:), (Expr(:., s) for s in sort!(filter!(Base.Fix1(Base.isexported, Stat), names(Stat,all=true)), by=(!=(:Statistics))))...)))
```

**Comment by vtjnash** [comment]
> And is equivalent to

It seems that it should be equivalent to this, but there is a warning that seems unintended with attempting it:

```julia
julia> using Statistics: Statistics as Stat

julia> global Statistics

julia> using .Stat
WARNING: using Statistics.Statistics in module Main conflicts with an existing identifier.
```

**Comment by LilithHafner** [comment]
@vtjnash, am I correct in thinking that the `using` syntax does not load names by explicitly enumerating the exported names of the loaded module but instead simply adds the name of the loaded module to the list of modules used by the parent, done here:

https://github.com/JuliaLang/julia/blob/5643c601f4c9819ed055c111a2d7d14cff6c5f7c/src/module.c#L636

**Comment by vtjnash** [comment]
Apparently the history of this warning is https://github.com/JuliaLang/julia/issues/4715, so this might specifically need to avoid the warning there when using the `.Stat` module tries to bring it its own name (from the implicit `export Statistics`)

**Comment by LilithHafner** [comment]
Declaring a global is also an observable property and I don't think it makes sense to declare a global named `Statistics` on `using Statistics as Stat`

```juli
julia> Statistics
ERROR: UndefVarError: `Statistics` not defined in `Main`
Suggestion: check for spelling errors or missing imports.

julia> using Statistics: Statistics as Stat

julia> global Statistics

julia> using .Stat
WARNING: using Statistics.Statistics in module Main conflicts with an existing identifier.

julia> Statistics
ERROR: UndefVarError: `Statistics` not defined in `Main`
Suggestion: add an appropriate import or assignment. This global was declared but not assigned.
Hint: a global variable of this name also exists in Statistics.

julia> names(Main)
8-element Vector{Symbol}:
 :Base
 :Core
 :InteractiveUtils
 :Main
 :Statistics
```

**Comment by DilumAluthge** [comment]
Just to clarify, what's the current behavior of `using Foo as Bar`?

**Comment by mbauman** [comment]
It's not valid syntax:

```
julia> :(using Foo as Bar)
ERROR: ParseError:
# Error @ REPL[1]:1:8
:(using Foo as Bar)
#      └─────────┘ ── `using` with `as` renaming requires a `:` and context module
Stacktrace:
 [1] top-level scope
   @ none:1
```

**Comment by LilithHafner** [comment]
The Julia object model does have the notion of "using all the exported symbols except for one". As I see it we have a few options if we want to implement this
1) Create an explicit binding for the module name when running `using Mod`, `using Mod as M`, or `using Mod.SubMod` (this is already [sometimes](https://github.com/JuliaLang/julia/issues/52817) the case) and stop implicitly exporting the module name itself (technically breaking but likely a minor change)
2) 1 but also add a special case to the `Base.isexported` function to make it's behavior stay the same (note: Base.isexported is not part of the public API of Julia <= 1.10)
3) When running `using Mod as M`, define a binding that shadows `Mod`
4) Make the object model more expressive

I'm currently investigating option 1. Open to alternative suggestions.

**Comment by fredrikekre** [comment]
Seems like a pretty niche usecase where you happily accept all `export`ed names from `using` the module but not the module name itself? What else does it offer over `using Foo; const F = Foo`? The `import`/`using` logic is already pretty complicated and something new users are often confused by, perhaps it isn't worth adding complexity? In particular not if it is breaking?

**Comment by LilithHafner** [comment]
The point is not so much to exclude loading the module name itself, but to support the syntax at all. I think this syntax is more intuitive and readable:

```julia
using LongPackageName as LPN
using LongPackageName; const LongPackageName = LPN
```

I do not think that the ability to rename packages on using will confuse new users because this renaming syntax is already supported in all other locations in using/import. The use case of wanting to load a long package under a short alias is reasonable in my opinion and may help allow folks to use descriptive package names without loosing brevity in domain specific usage.

If we are agreed that we want the `using LongPackageName as LPN` syntax, then we kinda have to make it not load `LongPackageName` which, sadly, does add some complexity.

This can be implemented in a technically non-breaking way (though I think the current PR is already practically non-breaking), the tradeoff there is that to make it technically non-breaking requires a more complex mental model and/or a more complex object model.

**Comment by fredrikekre** [comment]
> I think this syntax is more intuitive and readable:

I disagree; `using LongPackageName as LPN` feels like it should do what `import LongPackageName as LPN` does today.

**Comment by adienes** [comment]
xref https://github.com/JuliaLang/julia/issues/38521 (probably a dupe, technically)
